plugins/spu2-x/src/Windows/Config.cpp

```diff
@@ -294,6 +305,20 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
                     EndDialog(hWnd, 0);
                     break;
 
+                case IDC_OUTPUT:
+                    if (wmEvent == CBN_SELCHANGE) {
+                        const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
+                        bool IsConfigurable = module != 0;
+                        bool XAudio2Expansion = module == 1 || module == 4;
+
+                        EnableWindow(GetDlgItem(hWnd, IDC_OUTCONF), IsConfigurable);
+                        EnableWindow(GetDlgItem(hWnd, IDC_VOLUME_SLIDER), IsConfigurable);
+                        EnableWindow(GetDlgItem(hWnd, IDC_LATENCY_SLIDER), IsConfigurable);
+
+                        EnableWindow(GetDlgItem(hWnd, IDC_SPEAKERS), XAudio2Expansion);
```

Comment:
This looks like a copy of the above code. It might be a good idea to put the common stuff into a function.

Reply:
You are correct, it is a copy of the code above because the proper values need to be applied during init of the dialog and final output. Can you explain more about putting this into a function? It's already contained in a ```case WM_INITDIALOG``` and ```case IDC_OUTPUT``` in both instances.

Reply:
`case` isn't a function, it's a label.

By put in function I mean create a function like:
```
void common_code_function_please_name_well(HWND window)
{
    // common code here
}
```
and then call that function instead of having duplicated code.

Reply:
That's what I figured it was lol. I'll do some work on it when I get a chance in a bit.